blueocean-personalization/src/main/js/components/DashboardCards.jsx

```diff
@@ -56,188 +36,44 @@ export class DashboardCards extends Component {
             return null;
         }
 
         const favoriteCards = this.props.favorites.map(favorite => {
             const pipeline = favorite.item;
-            const latestRun = pipeline.latestRun;
-            const names = this._extractNames(pipeline);
-
-            let status = null;
-            let startTime = null;
-            let estimatedDuration = null;
-            let commitId = null;
-            let runId = null;
-
-            if (latestRun) {
-                if (latestRun.result) {
-                    status = latestRun.result === 'UNKNOWN' ? latestRun.state : latestRun.result;
-                }
-
-                startTime = latestRun.startTime;
-                estimatedDuration = latestRun.estimatedDurationInMillis;
-                commitId = latestRun.commitId;
-                runId = latestRun.id;
-            } else {
-                status = 'NOT_BUILT';
-            }
 
             return (
                 <div key={favorite._links.self.href}>
                     <PipelineCard
                       router={this.props.router}
-                      item={pipeline}
-                      status={status}
-                      startTime={startTime}
-                      estimatedDuration={estimatedDuration}
-                      organization={pipeline.organization}
-                      fullName={names.fullName}
-                      pipeline={names.pipelineName}
-                      branch={names.branchName}
-                      commitId={commitId}
-                      runId={runId}
+                      runnable={pipeline}
                       favorite
-                      onRunAgainClick={(pipeline1) => this._onRunAgainClick(pipeline1)}
-                      onRunClick={(pipeline2) => this._onRunClick(pipeline2)}
-                      onStopClick={(pipeline3) => this._onStopClick(pipeline3)}
                       onFavoriteToggle={(isFavorite) => this._onFavoriteToggle(isFavorite, favorite)}
                     />
                 </div>
             );
         });
 
         return (
-            <div className="favorites-card-stack">
-                <TransitionGroup transitionName="vertical-expand-collapse"
-                  transitionEnterTimeout={300}
-                  transitionLeaveTimeout={300}
-                >
-                    {favoriteCards}
-                </TransitionGroup>
-            </div>
-        );
-    }
-
-    render() {
-        return (
             <FavoritesProvider store={this.props.store}>
-                { this._renderCardStack() }
+                <div className="favorites-card-stack">
+                    <TransitionGroup transitionName="vertical-expand-collapse"
+                      transitionEnterTimeout={300}
+                      transitionLeaveTimeout={300}
+                    >
+                        {favoriteCards}
+                    </TransitionGroup>
+                </div>
             </FavoritesProvider>
         );
```

Comment:
Now that this component is just rendering a stack of cards the `render` method is a lot simpler.